internal/charmstore/store.go

```diff
@@ -505,6 +505,84 @@ func (s *Store) UpdateBaseEntity(url *charm.Reference, update interface{}) error
 	return nil
 }
 
+// Promulgate sets the base entity of url to be promulgated, and unsets promulgated
+// on any other base entity for charms with the same name. It also calculates the next
+// promulgated URL for the charms owned by the new owner and sets those charms.
+//
+// Note: This code is known to have some unfortunate (but not dangerous) race
```

Comment:
Let's enumerate the race conditions here and explain why they're not dangerous, please.